mesop/components/uploader/e2e/uploader_app.py

```diff
@@ -0,0 +1,39 @@
+import base64
+
+import mesop as me
+
+
+@me.stateclass
+class State:
+  name: str
+  size: int
+  mime_type: str
+  contents: str
+
+
+@me.page(path="/components/uploader/e2e/uploader_app")
+def app():
+  state = me.state(State)
+  me.uploader(
+    label="Upload Image",
+    accepted_file_types=["image/jpeg", "image/png"],
+    on_upload=handle_upload,
+  )
+
+  if state.contents:
+    with me.box(style=me.Style(margin=me.Margin.all(10))):
+      me.text(f"File name: {state.name}")
+      me.text(f"File size: {state.size}")
+      me.text(f"File type: {state.mime_type}")
+
+    with me.box(style=me.Style(margin=me.Margin.all(10))):
+      me.image(src=state.contents)
+
+
+def handle_upload(event: me.UploadEvent):
+  state = me.state(State)
+  for file in event.files:
```

Comment:
This is kind of funky, it's basically overwriting the same properties on state over again. I think this technically workers since this is just a single file upload right now, but it's kind of confusing as an example. Instead, I'd do something like assert that files is length 1 and only read the first element.

Reply:
Good point. I took your suggestion from the other comment to just use `file` for now.